src/Microsoft.AspNet.OData.Shared/EdmChangedObjectCollection.cs

```diff
@@ -57,5 +69,288 @@ private void Initialize(IEdmEntityType entityType)
             _edmTypeReference = new EdmCollectionTypeReference(_edmType);
         }
+
+        /// <summary>
+        /// Patch for Types without underlying CLR types
+        /// </summary>
+        /// <param name="originalCollection"></param>
+        /// <returns>ChangedObjectCollection response</returns>
+        public EdmChangedObjectCollection Patch(ICollection<EdmStructuredObject> originalCollection)
+        {            
+            PatchHandler = new DefaultTypelessPatchHandler(originalCollection, _entityType);
+
+            return CopyChangedValues();
+        }
+
+        /// <summary>
+        /// Handler for users Create, Get and Delete Methods
+        /// </summary>
+        internal TypelessPatchMethodHandler PatchHandler { get; set; }
+
+        /// <summary>
+        /// Patch for EdmChangedObjectCollection, a collection for IEdmChangedObject 
+        /// </summary>
+        /// <returns>ChangedObjectCollection response</returns>
+        public EdmChangedObjectCollection Patch(TypelessPatchMethodHandler patchHandler)
+        {
+            PatchHandler = patchHandler;
+            return CopyChangedValues();
+        }
+
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1800:DoNotCastUnnecessarily")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Maintainability", "CA1506:AvoidExcessiveClassCoupling")]
+        internal EdmChangedObjectCollection CopyChangedValues()
+        {
+            EdmChangedObjectCollection changedObjectCollection = new EdmChangedObjectCollection(_entityType);
+            List<IEdmStructuralProperty> keys = _entityType.Key().ToList();
+
+            foreach (IEdmChangedObject changedObj in Items)
+            {                
+                DataModificationOperationKind operation = DataModificationOperationKind.Update;
+                EdmStructuredObject originalObj = null;
+                string errorMessage = string.Empty;
+                string geterrorMessage = string.Empty;
+                IDictionary<string, object> keyValues = GetKeyValues(keys, changedObj);
+
+                try
+                {
+                    EdmStructuredObject original = null;
+                    EdmDeltaDeletedEntityObject deletedObj = changedObj as EdmDeltaDeletedEntityObject;
+
+                    if (deletedObj != null)
+                    {
+                        operation = DataModificationOperationKind.Delete;
+
+                        if (PatchHandler.TryDelete(keyValues, out errorMessage) != PatchStatus.Success)
+                        {
+                            //Handle Failed Operation - Delete                            
+                            PatchStatus status = PatchHandler.TryGet(keyValues, out original, out geterrorMessage);
+                            if (status == PatchStatus.Success)
```

Comment:
>PatchStatus.Success [](start = 42, length = 19)

what happens if status here is not success?  for the delete case, if the item has already been deleted (presumably TryDelete would return notfound?) then TryGet would also return not found, and the changedObjectCollection should have the delete. Open question what it should have if TryGet returned an error -- I would probably still assume the record didn't exist, and add the deleted record to the result, perhaps with an annotation to indicate that the status is uncertain.